object/slicer/slicer.go

```diff
@@ -164,20 +163,44 @@ func slice(ctx context.Context, ow ObjectWriter, header object.Object, data io.R
 	objectPayloadLimit := childPayloadSizeLimit(opts)
 
 	var n int
-	bChunk := opts.payloadBuffer
-	if bChunk == nil {
-		bChunk = make([]byte, objectPayloadLimit)
-	}
 
 	writer, err := initPayloadStream(ctx, ow, header, signer, opts)
 	if err != nil {
 		return rootID, fmt.Errorf("init writter: %w", err)
 	}
 
+	var buf []byte
+	var buffered uint64
+
 	for {
-		n, err = data.Read(bChunk)
+		buffered = writer.rootMeta.length
+		if writer.withSplit {
+			buffered = writer.childMeta.length
+		}
+
+		if buffered == objectPayloadLimit && uint64(len(writer.payloadBuffer)) <= objectPayloadLimit {
+			// in this case, the read buffers are exhausted, and it is unclear whether there
+			// will be more data. We need to know this right away, because the format of the
+			// final objects depends on it. So read to temp buffer
+			buf = []byte{0}
```

Comment:
what does this element mean?

Reply:
Looks like you're doing additional reads per-chunk while I thought it's possible to do it once per object.

Reply:
> after subsequent chunks, we need to know is it the last chunk or not

Why? You just pack another chunk and if the next read is `0, EOF` you create a link object.

Reply:
> You just pack another chunk

penult and last children have different formats - the latter contains the finalized parent header in its header while all previous ones dont

Reply:
> what element? If u mean single byte buffer, the comment tries to explain

so this is only a try to _read something more_? no problems with comments (at least for now), problems with understanding

Reply:
yes, read ahead